## Using functions from scripts

One can import classes and funtions from scripts in the same folder using the name of the file in question. In this ase the file is *tools.py*. THus we do the following to import them. It is also possible to set an alias for ease of use.

In [1]:
import tools as tls

The data cleaning functions are methods of the **data_cleaner** class object. Thus to use them we can create one such object and call the specific methods when they are needed.

In [2]:
cleaner = tls.data_cleaner()

### Getting climate factors 2013-2017

In [3]:
df_rain, df_temp, df_water, df_climate_factors = cleaner.climate_factors('raw data/WORLDBANK_rainfall.csv', 'raw data/WORLDBANK_temperature.csv', 'raw data/AQUASTAT_water_resources.csv')

In [4]:
df_climate_factors.to_csv('clean data/climate_factors.csv')
df_climate_factors

Country  Temperature (°C)  Total Rainfall (mm)  \
0       AFG         14.074742           349.736945   
1       AGO         22.182196           960.024065   
2       ALB         12.754647          1079.459168   
3       AND         12.402212           760.241065   
4       ARE         28.010773            64.449765   
..      ...               ...                  ...   
197     WSM         27.578074          3162.300825   
198     YEM         24.211854           161.796177   
199     ZAF         18.620716           403.002933   
200     ZMB         22.412762           895.648672   
201     ZWE         21.975148           516.969557   

     Total internal renewable water resources (IRWR)  \
0                                            47.1500   
1                                           148.0000   
2                                            26.9000   
3                                             0.3156   
4                                             0.1500   
..                                               ...   
197                                              NaN   
198                                           2.1000   
199                                          44.8000   
200                                          80.2000   
201                                          12.2600   

     Total external renewable water resources (ERWR)  \
0                                              18.18   
1                                               0.40   
2                                               3.30   
3                                                NaN   
4                                               0.00   
..                                               ...   
197                                             0.00   
198                                             0.00   
199                                             6.55   
200                                            24.60   
201                                             7.74   

     Total renewable water resources  Dependency ratio  \
0                            65.3300         28.722600   
1                           148.4000          0.269542   
2                            30.2000         10.927152   
3                             0.3156               NaN   
4                             0.1500          0.000000   
..                               ...               ...   
197                           0.0000          0.000000   
198                           2.1000          0.000000   
199                          51.3500         12.840467   
200                         104.8000         23.473282   
201                          20.0000         38.700000   

     Total exploitable water resources  
0                                  NaN  
1                                  NaN  
2                               13.000  
3                                  NaN  
4                                  NaN  
..                                 ...  
197                                NaN  
198                                NaN  
199                             11.972  
200                                NaN  
201                              1.500  

[202 rows x 8 columns]

### Getting water stress indicators 2013-2017

In [5]:
df_waterstress = cleaner.water_stress('raw data/AQUASTAT_water_stress.csv')

In [6]:
df_waterstress.to_csv('clean data/water_stress.csv')
df_waterstress

Variable Name Country  Water stress (MDG)  Water use efficiency (SDG)  \
0                 AFG           31.045462                    0.923778   
1                 AGO            0.475539                  142.467836   
2                 ALB            3.933775                    6.656907   
3                 ARE         1708.000000                   92.773763   
4                 ARG            4.301333                   13.616564   
..                ...                 ...                         ...   
175               VNM            9.259150                    2.349448   
176               YEM          169.761905                    5.219357   
177               ZAF           37.740993                   14.659097   
178               ZMB            1.500000                   12.764894   
179               ZWE           16.695000                    5.213329   

Variable Name  Water stress (SDG)  
0                       54.757019  
1                        1.871883  
2                        7.139423  
3                     1708.000000  
4                       10.456664  
..                            ...  
175                     18.130315  
176                    169.761905  
177                     62.055716  
178                      2.835498  
179                     31.346226  

[180 rows x 4 columns]

### Getting socio economic factors

In [7]:
df_aqua, df_unicef, df_socioec_factors = cleaner.socioecon_factors('raw data/aquastat_socio_economic.csv', 'raw data/unicef_socio_economic.csv')

In [8]:
df_socioec_factors.to_csv('clean data/socioec_factors.csv')
df_socioec_factors

Country GDP per capita (current US$/inhab)  \
0       AFG                         605.557362   
1       AGO                        4100.291004   
2       ALB                        4445.132198   
3       ARE                       40180.344045   
4       ARG                       14517.291248   
..      ...                                ...   
136     VNM                        2342.244006   
137     WSM                         4197.37637   
138     YEM                         854.926716   
139     ZAF                        6153.459018   
140     ZMB                         1513.27609   

    Human Development Index (HDI) [highest = 1] (-)  \
0                                             0.493   
1                                             0.576   
2                                             0.789   
3                                             0.864   
4                                             0.832   
..                                              ...   
136                                            0.69   
137                                           0.706   
138                                           0.463   
139                                           0.704   
140                                           0.589   

    Population density (inhab/km2) Rural population (1000 inhab)  \
0                    55.5955534111                     26558.609   
1                    23.9165524986                     10472.554   
2                   104.0089794446                      1190.155   
3                    133.584947902                      1292.709   
4                    15.8024528845                      3652.804   
..                             ...                           ...   
136                 285.6041059083                     61898.302   
137                   68.785915493                       160.194   
138                  52.7204557077                     18075.808   
139                  46.7641896825                     19369.002   
140                  22.3935358286                      9747.709   

    Rural population with access to safe drinking-water (JMP) (%)  \
0                                                 47.0              
1                                                 28.2              
2                                                 95.2              
3                                                100.0              
4                                                100.0              
..                                                 ...              
136                                               96.9              
137                                               99.3              
138                                               46.5              
139                                               81.4              
140                                               51.3              

    Total population with access to safe drinking-water (JMP) (%)  \
0                                                 55.3              
1                                                 49.0              
2                                                 95.1              
3                                                 99.6              
4                                                 99.1              
..                                                 ...              
136                                               97.6              
137                                               99.0              
138                                               54.9              
139                                               93.2              
140                                               65.4              

    Urban population (1000 inhab)  \
0                        8971.472   
1                       19311.639   
2                        1740.032   
3                        8107.436   
4                       40618.237   
..                            ...   
136     

## Checking countries

Need to see what country codes are shared across all datasets. The AQUA and Unicef datasets arleady have been cheked across each other. Tus can just use on of the two.

In [9]:
shared_countries = set.intersection(set(df_climate_factors['Country']), set(df_socioec_factors['Country']), set(df_waterstress['Country']))

In [10]:
len(shared_countries)

134

Thus we have 134 countries that are shared across the datasets. We will write a seperate *.csv* file that includes all these countries.

In [11]:
import pandas as pd
shared = pd.DataFrame(list(shared_countries))
shared.to_csv('shared_country_codes.csv')

### Getting projected temperatures - 2020-2039

In [12]:
df_temp_proj = cleaner.temp_proj('raw data/tas_2020_2039_mavg_rcp26_AFG_CAF.csv')

In [13]:
df_temp_proj.to_csv('clean data/projected_temp_2020-2039.csv')
df_temp_proj

Country  Temperature (°C)
0       AFG         12.369804
1       AGO         20.311982
2       ALB         14.062129
3       AND         12.352881
4       ARE         24.896902
..      ...               ...
188     XRK         11.533050
189     YEM         22.247077
190     ZAF         15.588705
191     ZMB         19.578873
192     ZWE         19.294114

[193 rows x 2 columns]